### PR TITLE
Add support for React Native 0.76

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

Updates the Fabric Example app to React Native 0.76
Adapts library to compile with React Native 0.76

(There are _some_ changes in `yarn.lock` due to me recreating the app from scratch, I was getting random babel errors, so there must have been some missed upgrade steps in the past and this seemed quicker than trying to finding the issue)

## Test plan

Test the Fabric Example app

### CODE REVIEW

File: android/src/main/jni/CMakeLists.txt [hunk 1]
```diff
@@ -26,12 +26,21 @@ target_include_directories(
...
+  target_link_libraries(
+    ${PACKAGE_NAME}
+    ReactAndroid::react_render_core
+    ReactAndroid::react_render_uimanager
+    ReactAndroid::react_render_graphics
+    ReactAndroid::jsi
+    ReactAndroid::react_nativemodule_core
+  )
+else ()
+  message(FATAL_ERROR "react-native-gesture-handler requires react-native 0.75 or newer.")
```
Comment (tomekzaw) [inline android/src/main/jni/CMakeLists.txt:45]:
```suggestion
  message(FATAL_ERROR "react-native-gesture-handler on new architecture requires react-native 0.75 or newer.")
```

File: android/src/main/jni/CMakeLists.txt [hunk 2]
```diff
@@ -26,12 +26,21 @@ target_include_directories(
...
+endif ()
```
Comment (tomekzaw) [inline android/src/main/jni/CMakeLists.txt:46]:
```suggestion
endif ()

```

File: android/src/main/jni/CMakeLists.txt [hunk 3]
```diff
@@ -26,12 +26,21 @@ target_include_directories(
 )
 
 find_package(ReactAndroid REQUIRED CONFIG)
-
+if (ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
 target_link_libraries(
```
Comment (tomekzaw) [inline android/src/main/jni/CMakeLists.txt:30]: Let's add two spaces of indent here

File: FabricExample/package.json [hunk 4]
```diff
@@ -45,4 +43,4 @@
     "node": ">=18"
   },
   "packageManager": "yarn@1.22.22"
-}
+}
```
Comment (tomekzaw) [inline FabricExample/package.json:46]:
```suggestion
}

```

**Review by tomekzaw — APPROVED** [review]

File: android/src/main/jni/CMakeLists.txt (hunk 3 above)
Comment (j-piasecki) [reply android/src/main/jni/CMakeLists.txt]: Done in [be14680](https://github.com/software-mansion/react-native-gesture-handler/pull/3109/commits/be146807078a1126e5956bd481f213495c519899)

**Review by tomekzaw — APPROVED** [review]
👏

**Pull request merged**